engine/table/src/main/java/io/deephaven/engine/table/impl/select/PatternFilter.java

```diff
@@ -0,0 +1,193 @@
+    }
+
+    @Override
+    public void init(TableDefinition tableDefinition) {
+        final ColumnDefinition<?> column = tableDefinition.getColumn(columnName.name());
+        if (column == null) {
+            throw new RuntimeException(String.format("Column '%s' doesn't exist in this table, available columns: %s",
+                    columnName.name(), tableDefinition.getColumnNames()));
+        }
+    }
+
+    @Override
+    public WritableRowSet filter(RowSet selection, RowSet fullSet, Table table, boolean usePrev) {
+        final ColumnSource<String> columnSource = table.getColumnSource(columnName.name());
+        return ChunkFilter.applyChunkFilter(selection, columnSource, usePrev, this);
+    }
+
+    @Override
+    public boolean isSimpleFilter() {
+        return true;
+    }
+
+    @Override
+    public void setRecomputeListener(RecomputeListener listener) {
+
+    }
+
+    @Override
+    public boolean canMemoize() {
+        return true;
+    }
+
+    @Override
+    public List<String> getColumns() {
+        return Collections.singletonList(columnName.name());
+    }
+
+    @Override
+    public List<String> getColumnArrays() {
+        return Collections.emptyList();
+    }
+
+    @Override
+    public WhereFilter copy() {
+        return new PatternFilter(columnName, pattern, mode, invertPattern);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o)
+            return true;
+        if (o == null || getClass() != o.getClass())
+            return false;
+        PatternFilter that = (PatternFilter) o;
+        if (invertPattern != that.invertPattern)
+            return false;
+        if (!columnName.equals(that.columnName))
+            return false;
+        if (!patternEquals(pattern, that.pattern))
+            return false;
+        return mode == that.mode;
+    }
+
+    @Override
+    public int hashCode() {
+        int result = columnName.hashCode();
+        result = 31 * result + patternHashcode(pattern);
+        result = 31 * result + mode.hashCode();
+        result = 31 * result + (invertPattern ? 1 : 0);
+        return result;
+    }
+
+    @Override
+    public String toString() {
+        return "PatternFilter{" +
+                "columnName=" + columnName +
+                ", pattern=" + pattern.pattern() + " " + pattern.flags() +
+                ", mode=" + mode +
+                ", invertPattern=" + invertPattern +
+                '}';
+    }
+
+    @Override
+    public void filter(ObjectChunk<String, ? extends Values> values, LongChunk<OrderedRowKeys> keys,
+            WritableLongChunk<OrderedRowKeys> results) {
+        results.setSize(0);
+        for (int ix = 0; ix < values.size(); ++ix) {
+            if (matcher(values.get(ix))) {
+                results.add(keys.get(ix));
+            }
+        }
+    }
+
+    private boolean matcher(String value) {
+        return value != null && (invertPattern ^ mode.test(pattern, value));
```

Comment:
This seems fine. I would probably have elevated this up to 4 different static filter methods (for each combination of invertPattern and mode), and just delegated to the right one.